js/src/modal.js

```diff
@@ -307,8 +309,22 @@ class Modal {
     if (this._isShown && this._config.keyboard) {
       EventHandler.on(this._element, Event.KEYDOWN_DISMISS, event => {
         if (event.which === ESCAPE_KEYCODE) {
-          event.preventDefault()
-          this.hide()
+          if (this._config.backdrop === 'static') {
+            const hideEvent = EventHandler.trigger(this._element, Event.HIDE_PREVENTED)
```

Comment:
Can you create a separate function for your `if` condition, btw it seems it's the same code used a few lines later, so it should be better to just call your function two times

Reply:
Sure thing! 👍